mysqld_exporter.go

```diff
@@ -178,7 +182,18 @@ func main() {
 			enabledScrapers = append(enabledScrapers, scraper)
 		}
 	}
-	http.HandleFunc(*metricPath, prometheus.InstrumentHandlerFunc("metrics", newHandler(enabledScrapers)))
+	http.HandleFunc(*metricPath, prometheus.InstrumentHandlerFunc("metrics", newHandler(enabledScrapers, &health)))
+	// Add an endpoint for mysql health state.
+	http.HandleFunc("/-/mysqld_healthy", func(w http.ResponseWriter, r *http.Request) {
+		if err := health.Get(); err != nil {
+			w.WriteHeader(http.StatusServiceUnavailable)
+			w.Write([]byte("critical: mysql_up"))
+			fmt.Fprint(w, err)
```

Comment:
The above three lines can be combined to:

```go
http.Error(w, fmt.Sprintf("critical: mysql_up: %v", err), http.StatusServiceUnavailable)
```

(currently I think there would be no space or newline before the `err` message)